SpockOpaleye/src/UserApi.hs

```diff
@@ -0,0 +1,170 @@
+{-# LANGUAGE Arrows #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module UserApi
+  ( create_user
+  , read_users
+  , read_user_by_id
+  , read_users_for_tenant
+  , add_role_to_user
+  , remove_role_from_user
+  , update_user
+  , remove_user
+  , activate_user
+  ) where
+
+import Control.Arrow
+import Data.Text
+import Database.PostgreSQL.Simple (Connection)
+import DataTypes
+import GHC.Int
+import Opaleye
+import OpaleyeDef
+
+create_user :: Connection -> User -> IO (Maybe User)
+create_user conn user@User {user_id = _
+                           ,user_tenantid = tenant_id
+                           ,user_username = username
+                           ,user_password = password
+                           ,user_firstname = first_name
+                           ,user_lastname = last_name
+                           ,user_status = status} = do
+  ids <-
+    runInsertManyReturning
+      conn
+      userTable
+      (return
+         ( Nothing
+         , constant tenant_id
+         , pgStrictText username
+         , pgStrictText password
```

Comment:
Yes. I am aware of this and have lined it up in things to take care before or in the next sprint.